### PR TITLE
iPXE provisioning test

### PR DESCRIPTION
**Pull request by shubhamsg199**
Adding iPXE provisioning test as a part of the component audit.

### CODE REVIEW

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_provisioning.py -k test_rhel_ipxe_provisioning
provisioning: true

File: tests/foreman/api/test_provisioning.py [hunk 1]
```diff
@@ -147,3 +148,123 @@ def test_rhel_pxe_provisioning(
...
+        2. Host installs right version of RHEL
+        3. Satellite is able to run REX job on the host
+        4. Host is registered to Satellite and subscription status is 'Success'
+
+    :parametrized: yes
+    """
+    # TODO: parametrize iPXE Chain BIOS as pxe loader after #BZ:2171172 is fixed
+    sat = module_provisioning_sat.sat
+    # set http url
+    ipxe_http_url = sat.install(
```
Comment (omkarkhatavkar) [inline tests/foreman/api/test_provisioning.py:190]: Do you mean `sat.execute`?

File: tests/foreman/api/test_provisioning.py [hunk 2]
```diff
@@ -147,3 +148,123 @@ def test_rhel_pxe_provisioning(
...
+    expected_rhel_version = f'{host_os.major}.{host_os.minor}'
+    host_ssh_os = module_provisioning_sat.sat.execute(
+        f'sshpass -p {settings.provisioning.host_root_password} '
+        'ssh -o StrictHostKeyChecking=no -o PubkeyAuthentication=no -o PasswordAuthentication=yes '
+        f'-o UserKnownHostsFile=/dev/null root@{provisioning_host.hostname} cat /etc/redhat-release'
+    )
+    assert host_ssh_os.status == 0
+    assert (
+        expected_rhel_version in host_ssh_os.stdout
+    ), 'Different than the expected OS version was installed'
```
Comment (omkarkhatavkar) [inline tests/foreman/api/test_provisioning.py:250]: `The installed OS version differs from the expected version '${expected_rhel_version}'.`

File: tests/foreman/api/test_provisioning.py (hunk 1 above)
Comment (shubhamsg199) [reply tests/foreman/api/test_provisioning.py:190]: No, sat.execute doesn't work here . I get `TypeError: Expected bytes, got InstallerCommand`

File: tests/foreman/api/test_provisioning.py [hunk 3]
```diff
@@ -147,3 +148,123 @@ def test_rhel_pxe_provisioning(
...
+    # Host is not blank anymore
+    provisioning_host.blank = False
+
+    # Wait for the host to be rebooted and SSH daemon to be started.
+    provisioning_host.wait_for_connection()
+
+    # Perform version check and check if root password is properly updated
+    host_os = host.operatingsystem.read()
+    expected_rhel_version = f'{host_os.major}.{host_os.minor}'
+    host_ssh_os = module_provisioning_sat.sat.execute(
```
[the remark this answers is not in the document]
Comment (shubhamsg199) [reply tests/foreman/api/test_provisioning.py]: Updated.

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_provisioning.py -k test_rhel_ipxe_provisioning
provisioning: true

File: tests/foreman/api/test_provisioning.py [hunk 4]
```diff
@@ -147,3 +148,123 @@ def test_rhel_pxe_provisioning(
     # assert that the host is subscribed and consumes
     # subsctiption provided by the activation key
     assert provisioning_host.subscribed, 'Host is not subscribed'
+
+
+@pytest.mark.e2e
+@pytest.mark.parametrize('pxe_loader', ['ipxe'], indirect=True)
+@pytest.mark.on_premises_provisioning
+@pytest.mark.rhel_ver_match('[8]')
+def test_rhel_ipxe_provisioning(
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_provisioning.py:157]: Can we extend this for all available rhel versions?

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_provisioning.py -k test_rhel_ipxe_provisioning
provisioning: true

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_provisioning.py -k test_rhel_ipxe_provisioning
provisioning: true

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_provisioning.py -k test_rhel_ipxe_provisioning
provisioning: true

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_provisioning.py -k test_rhel_ipxe_provisioning
provisioning: true

**Review by Gauravtalreja1 — APPROVED** [review]

**Pull request merged**